solr/core/src/java/org/apache/solr/security/AuthenticationPlugin.java

```diff
@@ -184,7 +181,7 @@ public Category getCategory() {
 
   @Override
   public MetricRegistry getMetricRegistry() {
-    return registry;
+    return metrics == null ? null : metrics.getMetricRegistry();
```

Comment:
Good point, we can.

Reply:
Actually, we can't because we already have a default with the same signature in `SolrInfoBean`, but I can improve the implementation there to avoid this repetition.